mmdeploy/codebase/mmseg/deploy/segmentation.py

```diff
@@ -268,5 +268,10 @@ def get_model_name(self) -> str:
         """
         assert 'decode_head' in self.model_cfg.model, 'model config contains'
         ' no decode_head'
-        name = self.model_cfg.model.decode_head.type[:-4].lower()
+        if isinstance(self.model_cfg.model.decode_head, list):
+            name = self.model_cfg.model.decode_head[-1].type[:-4].lower()
+        elif 'type' in self.model_cfg.model.decode_head:
+            name = self.model_cfg.model.decode_head.type[:-4].lower()
+        else:
+            raise NotImplementedError('Could not get the name of the model')
```

Comment:
Can SDK use the model without a name? If it could, can we give it a default name instead of raising error?

Reply:
@lvhan028

Reply:
@lvhan028

Reply:
Is this `name` related to the `component` defined in the SDK model? The necessary attributes of MMSegmentation's postprocessing module in SDK Model are defined as,
```json
{
      "type": "Task",
      "module": "mmseg",
      "name": "postprocess",
      "component": "ResizeMask",
      "params": {
          "num_classes": 19
      },
      "output": [
          "post_output"
      ],
      "input": [
          "prep_output",
          "infer_output"
      ]
  }
```

Reply:
Hi, check [2.1.1](https://openmmlab.feishu.cn/docs/doccn6J8vUdEMx4AkQNIuRlFqte) out. If it is useless, the attributes `get_model_name` can be removed then.